### DISCUSSION TITLE
Question for Permissions Check by_ssh

### DISCUSSION THREAD

**Discussion by CristianGLepore** [Linuxfabrik/monitoring-plugins#699] [General]
Hi!

I am trying to generate a check to validate the permissions of a directory in Ubuntu. For this, I am using the by_ssh plugin. What I need to validate is that a directory has 777 permissions, for this I am executing the command
`ls -l /`
Reviewing the documentation, I could see that there are parameters to indicate a notification when a regex is found within the response. Is there a way to give a notification when the answer is NOT rwxrwxrwx ?

**Comment by markuslf** [comment]
Sure. To achieve the desired behavior, you can use a negative lookahead in your Python regular expression. The negative lookahead allows you to specify a pattern that must not be present immediately after the match.

For example, you might want to match the directory "opt" in your output only if the entire line does not begin with "drwxrwxrwx".

A working cmdline call would be
`./by-ssh --hostname myhost --username myuser --warning-regex '^(?!drwxrwxrwx.*).*opt$' --command 'ls -ld /opt'`

**Comment by markuslf** [comment]
Another more consistent way using `stat` and a more readable regex (but same logic):
`./by-ssh --hostname myhost --username myuser --warning-regex '^(?!777)\d{3}$' --command 'stat /opt -c %a'`
